cads_adaptors/tools/area_selector.py

```diff
@@ -122,67 +135,70 @@ def get_dim_slices(
 
     # A final check that there is at least an overlap
     if not points_inside_range([start, end], coord_range):
-        incompatible_area_error(
-            dim_key, start, end, coord_range, context, thisError=NotImplementedError
-        )
+        incompatible_area_error(dim_key, start, end, coord_range, context)
 
     return [slice(start, end)]
 
 
 def area_selector(
-    infile: str,
+    ds: xr.Dataset,
+    area: list[float | int] | dict[str, float | int] = [+90, -180, -90, +180],
```

Comment:
Mutable default kwargs can be dangerous, as they can be unexpectedly changed, causing a different default value to be used the next time the function is called.

You should define a `DEFAULT_AREA` constant somewhere, and then change this to:

```suggestion
    area: list[float | int] | dict[str, float | int] = None,
```

Then in your function somewhere you could have something like:

```python
if area is None:
    area = DEFAULT_AREA
```

Reply:
Thanks, but having thought about it I will change to an arg. This function should only be called if an area is provided, a default value may lead unintended/unexpected results. From the DSS, it is only ever invoked when the area is provided, but I will make sure there are not any other implications, and make sure that it is provided in the correct position when called elsewhere

Reply:
This method is only called from one place (line 273), and I have updated that to be an positional arg

Reply:
Also, prior to this PR, the default AREA was incorrectly defined, demonstrating that it was never actually used anyway:
https://github.com/ecmwf-projects/cads-adaptors/pull/262/files#diff-463d10e8092658ca163578f7d529dc6ce5d329620a3483e62412a27529acbc2fL135